style: Unread counter can now display five digits

krlade opened the pull request:
* Unread counter could previously display only two digits which wasn't really useful, so I've added more space so it can now display five digits.
Before
![before](https://user-images.githubusercontent.com/110191114/224182913-7f2ecfe7-15da-49fd-bb84-f218663e01c7.png)
After
![after](https://user-images.githubusercontent.com/110191114/224182925-bec37461-68d1-4ee3-8a0d-85af17a33b2a.png)

* Do not add sender css classes for outgoing message, because it make the contrast really weak.
![example](https://user-images.githubusercontent.com/110191114/224182951-c780445e-cd35-4b80-bce1-3f8fac9ec976.png)

melix99 commented on src/session/content/message_row/bubble.rs:
```diff
@@ -284,15 +284,18 @@ impl MessageBubble {
-        imp.sender_label.add_css_class(color_class);
-        imp.sender_color_class.replace(Some(color_class.into()));
+        //Add sender css class only if message is incoming
+        if !self.has_css_class("outgoing") {
```
It's much better checking for the actual outgoing Message property instead of a generic css class.

melix99 commented on data/resources/ui/content-chat-history.ui:
```diff
@@ -68,6 +68,8 @@
                       <object class="GtkButton">
                         <property name="halign">center</property>
                         <property name="valign">end</property>
+                        <property name="margin-start">3</property>
```
Please use CSS instead. There's already a CSS class defined.

melix99 commented on data/resources/style.css:
```diff
@@ -131,7 +131,7 @@ sidebarsearchitemrow {
   background-color: @window_bg_color;
   box-shadow: inset 0 0 0 1px @borders;
-  margin: 6px;
+  margin: 6px 9px 6px 9px;
```
```suggestion
  margin: 6px 9px;
```

krlade committed:
style: Unread counter can now display five digits

Unread counter could previously display only two digits, which wasn't much
useful.

melix99 commented on src/session/content/message_row/bubble.rs:
```diff
@@ -218,7 +218,8 @@ impl MessageBubble {
 
-            self.update_sender_color(maybe_id);
+            //Color sender name only if message is incoming
+            self.update_sender_color(maybe_id, !message.is_outgoing());
```
Mh, why not just setting `show_sender` to `None` when the message is outgoing instead of doing this?

krlade replied:
The message is indeed outgoing, but that doesn't mean you're the sender of the message. E.g. you wrote a post in the channel and it's visible as an outgoing message in the discussion group, in that case we want to show sender name even though the message is outgoing.

melix99 replied:
Ah wait, I meant to set the first parameter of `update_sender_color` (`maybe_id`) to `None` when the message is outgoing.

melix99 commented on src/session/content/message_row/bubble.rs:
```diff
@@ -284,15 +287,11 @@ impl MessageBubble {
+            let color_class = SENDER_COLOR_CLASSES[sender_id as usize % SENDER_COLOR_CLASSES.len()];
+            imp.sender_label.add_css_class(color_class);
+            imp.sender_color_class.replace(Some(color_class.into()));
+        }
```
As I said in the group, this won't work because now the default sender color is not set anymore.

Anyway, I decided to make bigger changes to sender colors, which I'll work on another day. So, can you drop this commit for now?

melix99 commented on data/com.github.melix99.telegrand.metainfo.xml.in.in:
```diff
@@ -27,6 +27,13 @@
+  <custom>
+    <value key="Purism::form_factor">workstation</value>
+    <value key="Purism::form_factor">mobile</value>
+  </custom>
```
Mh, I think these aren't needed anymore these days? Why do we need them?

krlade replied:
Not sure, telegrand is currently displayed as desktop only in gnome software, but maybe it's a bug of the ci or maybe it's because not being added to the flathub/any other remote.
![Screenshot from 2023-03-28 00-10-18](https://user-images.githubusercontent.com/110191114/228079288-87a4c963-0630-4280-97ee-2fd6e3f82332.png)

krlade replied:
I should investigate it before commiting this
